backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/KeyValueHandler.java

```diff
@@ -249,4 +256,66 @@ public KeyValueHandler(Executor executor, FutureJdbi jdbi, String fieldType, Str
             executor);
     return currentFuture;
   }
+
+  public InternalFuture<MapSubtypes<KeyValue>> getExperimentRunHyperparameterConfigBlobMap(
+      List<String> expRunIds, List<String> selfAllowedRepositoryIds) {
+    return jdbi.withHandle(
+            handle -> {
+              String queryStr =
+                  "SELECT distinct vme.experiment_run_id, hecb.name, hecb.value_type, hecb.int_value, hecb.float_value, hecb.string_value  FROM hyperparameter_element_config_blob hecb "
+                      + "INNER JOIN config_blob cb ON cb.hyperparameter_element_config_blob_hash = hecb.blob_hash "
+                      + "INNER JOIN versioning_modeldb_entity_mapping vme ON vme.blob_hash = cb.blob_hash "
+                      + "WHERE cb.hyperparameter_type = :hyperparameterType AND vme.experiment_run_id IN (<expRunIds>) ";
+
+              if (config.populateConnectionsBasedOnPrivileges) {
+                if (selfAllowedRepositoryIds == null || selfAllowedRepositoryIds.isEmpty()) {
+                  return new ArrayList<AbstractMap.SimpleEntry<String, KeyValue>>();
+                } else {
+                  queryStr = queryStr + " AND vme.repository_id IN (<repoIds>)";
+                }
+              }
+
+              var query = handle.createQuery(queryStr);
+              query.bind("hyperparameterType", HYPERPARAMETER);
```

Comment:
This is a `KeyValueHandler`. Don't hardcode hyperparameter things. If you want, extract it to a hyperparameter handler that inherits from this one

Reply:
Extracted in separate hyperparameter config handler